app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -64,12 +66,12 @@ def feeds
     end
 
     def machine_id(feed)
-      os_resource_for(feed).properties['Machine Id']
+      os_resource_for(hawk_escape_id(feed)).properties['Machine Id']
     end
 
     def os_resource_for(feed)
       with_provider_connection do |connection|
-        os = os_for(feed)
+        os = os_for(hawk_escape_id(feed))
```

Comment:
You are right, it is. Good catch. I didn't test it because I don't know why, actually @jpkrohling could you please help me with how to verify that I didn't break your "association to machine" stuff?


> Wouldn't it be better to escape only prior to a server call..

done

> .. or modify the client to escape it?

It was done that way originally in the ruby client. Issue with it was that ids in canonical paths were already encoded so if one wants to reuse the path (that appeared in some JSON payload) as another param to another call, the client would encode it again (if it's part of the client logic). I am open to any better solution, but I didn't want to introduce any "unencode" functions, because it would be even more complicated.

All these ids that contains spaces, slashes, equals and other funky chars was a bad design decision (IMO). Now there are 2 layers that are causing it. 1. is the agent that uses those chars that needs encoding 2. is the inventory that returns a data in JSON that has the id unencoded and the canonical paths encoded (example: https://git.io/v6UG8)